ui/app/src/Utils/Match.elm

```diff
@@ -1,9 +1,32 @@
-module Utils.Match exposing (consecutiveChars, jaro, jaroWinkler)
+module Utils.Match exposing (consecutiveChars, filter, jaro, jaroWinkler)
 
 import Char
+import Regex
 import Utils.List exposing (zip)
 
 
+filter : String -> List { a | name : String } -> List { a | name : String }
+filter search list =
+    let
+        re =
+            "(.+)?"
+                ++ (String.split "" search |> String.join "(.+)?")
+                ++ "(.+)?"
+                |> Regex.fromString
+                |> Maybe.withDefault Regex.never
```

Comment:
Any recommendation on how to make this less messy is greatly appreciated :)

Reply:
Maybe this would read nicer:

```
("(.+)?" ++ String.join "(.+)?" (String.split "" search) ++ "(.+)?")
```